### ISSUE TITLE
HTML renderer doesn't preserve whitespace in text, while other renderers do

### ISSUE THREAD

**Issue by andrewleader** [Microsoft/AdaptiveCards#1984]
# Platform

* JavaScript

# Author

Author, sending cards to hosts that use JavaScript renderer

# Version of SDK

1.0

# Issue

The JavaScript renderer doesn't preserve whitespace within TextBlock text. Other renderers, like UWP, do. This is a consistency issue, and also blocks proposals like #1078 working for monospace blocks (since those need whitespace preserved).

Note that this may be a side effect of the Markdown-It processor used in the JavaScript renderer on the Visualizer?

## Repro payload

```json
{
    "type": "AdaptiveCard",
    "body": [
        {
            "type": "TextBlock",
            "text": "Tool    started..."
        }
    ],
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.0"
}
```

## As seen on web Visualizer...
Notice the extra spaces were removed

![image](https://user-images.githubusercontent.com/13246069/46319716-fc60d600-c58f-11e8-9981-ccfdf56a2cae.png)

## As seen with UWP renderer...
Notice the spaces are preserved

![image](https://user-images.githubusercontent.com/13246069/46319747-1995a480-c590-11e8-9b52-61632da5e5aa.png)

**Comment by dclaux** [comment]
This is not a problem with the JS renderer. It's a markdown issue, and is actually by design as far as I can tell. Use `&nbsp;` to prevent collapsing of spaces.

**Comment by andrewleader** [comment]
@dclaux yep I'm discovering that, we've got a whole lot of Markdown inconsistencies around text (as I've renamed the issue). JavaScript seems to handle it correctly (via use of the Markdown-It).

Can't use `&nbsp;` since when rendered on UWP or other renderers, it renders exactly `&nbsp;` (hence the added additional details... there's also inconsistency around new lines, etc... fun fun!).
